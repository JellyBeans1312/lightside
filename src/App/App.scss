
@font-face {font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff2") format("woff2"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS") format("svg");
}

html {
  height: 100%;
  overflow: hidden;
  box-sizing: border-box;
}

body {
  background: url('../images/star-background.jpg');
  background-size: cover;
  background-attachment: fixed;
  position: absolute;
  width: 100vw;
  height: 100%;
}

#root {
  height: 100%;
}

.App {
  text-align: center;
  height: 100%;
}

nav {
  box-sizing: border-box;
  padding-top: 20px;
  // border: white 1px solid;
  width: 100%;
  height: 15%;
  display: flex;
  flex-wrap: wrap;
  h1 {
    color: white;
    width: 300px;
    margin: auto;
    font-size: 50px;
  }
  div {
    width: 10%;
  }
  section {
    width: 80%;
    margin: auto;
    display: flex;
    justify-content: space-around;
    button {
      height: 30px;
      width: 100px;
    }
  }
}

.link {
  height: 20px;
  margin-top: 10px;
}

h2 {
  margin: 0;
}

h1,
span,
h2,
.nav {
  font-family: "STARWARS";
  text-decoration-color: white
}

h1:hover,
span:hover,
h2:hover,
.nav:hover{
  color: #ffd700;
  text-decoration-color: #ffd700
}

.home {
  width: 100%;
  background: none;
  text-decoration: none;
}

.active {
  h1,
  h2 {
    color:#ffd700
  }
}
