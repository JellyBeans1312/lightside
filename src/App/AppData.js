const AppData = {

  fetchedPeople() {
    return({
      "count": 87, 
      "next": "https://swapi.co/api/people/?page=2", 
      "previous": null, 
      "results": [
          {
              "name": "Luke Skywalker", 
              "height": "172", 
              "mass": "77", 
              "hair_color": "blond", 
              "skin_color": "fair", 
              "eye_color": "blue", 
              "birth_year": "19BBY", 
              "gender": "male", 
              "homeworld": "https://swapi.co/api/planets/1/", 
              "films": [
                  "https://swapi.co/api/films/2/", 
                  "https://swapi.co/api/films/6/", 
                  "https://swapi.co/api/films/3/", 
                  "https://swapi.co/api/films/1/", 
                  "https://swapi.co/api/films/7/"
              ], 
              "species": [
                  "https://swapi.co/api/species/1/"
              ], 
              "vehicles": [
                  "https://swapi.co/api/vehicles/14/", 
                  "https://swapi.co/api/vehicles/30/"
              ], 
              "starships": [
                  "https://swapi.co/api/starships/12/", 
                  "https://swapi.co/api/starships/22/"
              ], 
              "created": "2014-12-09T13:50:51.644000Z", 
              "edited": "2014-12-20T21:17:56.891000Z", 
              "url": "https://swapi.co/api/people/1/"
          }]
        })
  },

  unfilteredPeople() {
    return {birth_year: "19BBY",
    created: "2014-12-09T13:50:51.644000Z",
    edited: "2014-12-20T21:17:56.891000Z",
    eye_color: "blue",
    films: ["https://swapi.co/api/films/2/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/", "https://swapi.co/api/films/7/"],
    gender: "male",
    hair_color: "blond",
    height: "172",
    homeworld: "https://swapi.co/api/planets/1/",
    mass: "77",
    name: "Luke Skywalker",
    skin_color: "fair",
    species: ["https://swapi.co/api/species/1/"],
    starships: ["https://swapi.co/api/starships/12/", "https://swapi.co/api/starships/22/"],
    url: "https://swapi.co/api/people/1/",
    vehicles: ["https://swapi.co/api/vehicles/14/", "https://swapi.co/api/vehicles/30/"]}
  },

  filteredPeople() {
    return ["Luke Skywalker", "Birth Year: 19BBY", "Gender: male", "Height: 172", "Eye Color: blue", "2014-12-09T13:50:51.644000Z", false]
  },

  fetchedPlanets() {
    return{
      "count": 61, 
      "next": "https://swapi.co/api/planets/?page=2", 
      "previous": null, 
      "results": [
          {
              "name": "Alderaan", 
              "rotation_period": "24", 
              "orbital_period": "364", 
              "diameter": "12500", 
              "climate": "temperate", 
              "gravity": "1 standard", 
              "terrain": "grasslands, mountains", 
              "surface_water": "40", 
              "population": "2000000000", 
              "residents": [
                  "https://swapi.co/api/people/5/", 
                  "https://swapi.co/api/people/68/", 
                  "https://swapi.co/api/people/81/"
              ], 
              "films": [
                  "https://swapi.co/api/films/6/", 
                  "https://swapi.co/api/films/1/"
              ], 
              "created": "2014-12-10T11:35:48.479000Z", 
              "edited": "2014-12-20T20:58:18.420000Z", 
              "url": "https://swapi.co/api/planets/2/"
          }]}
  },

  unfilteredPlanets() {
    return {climate: "temperate",
    created: "2014-12-10T11:35:48.479000Z",
    diameter: "12500",
    edited: "2014-12-20T20:58:18.420000Z",
    films: ["https://swapi.co/api/films/6/", "https://swapi.co/api/films/1/"],
    gravity: "1 standard",
    name: "Alderaan",
    orbital_period: "364",
    population: "2000000000",
    residents: ["https://swapi.co/api/people/5/", "https://swapi.co/api/people/68/", "https://swapi.co/api/people/81/"],
    rotation_period: "24",
    surface_water: "40",
    terrain: "grasslands, mountains",
    url: "https://swapi.co/api/planets/2/"}
  },

  filteredPlanets() {
    return ["Alderaan", "Terrain: grasslands, mountains", "Diameter: 12500", "Population: 2000000000", null, "2014-12-10T11:35:48.479000Z", false]
  },

  fetchedVehicles() {
    return {
      "count": 39, 
      "next": "https://swapi.co/api/vehicles/?page=2", 
      "previous": null, 
      "results": [
          {
              "name": "Sand Crawler", 
              "model": "Digger Crawler", 
              "manufacturer": "Corellia Mining Corporation", 
              "cost_in_credits": "150000", 
              "length": "36.8", 
              "max_atmosphering_speed": "30", 
              "crew": "46", 
              "passengers": "30", 
              "cargo_capacity": "50000", 
              "consumables": "2 months", 
              "vehicle_class": "wheeled", 
              "pilots": [], 
              "films": [
                  "https://swapi.co/api/films/5/", 
                  "https://swapi.co/api/films/1/"
              ], 
              "created": "2014-12-10T15:36:25.724000Z", 
              "edited": "2014-12-22T18:21:15.523587Z", 
              "url": "https://swapi.co/api/vehicles/4/"
          }]
    }
  },

  unfilteredVehicles() {
    return {cargo_capacity: "50000",
    consumables: "2 months",
    cost_in_credits: "150000",
    created: "2014-12-10T15:36:25.724000Z",
    crew: "46",
    edited: "2014-12-22T18:21:15.523587Z",
    films: ["https://swapi.co/api/films/5/", "https://swapi.co/api/films/1/"],
    length: "36.8",
    manufacturer: "Corellia Mining Corporation",
    max_atmosphering_speed: "30",
    model: "Digger Crawler",
    name: "Sand Crawler",
    passengers: "30",
    pilots: [],
    url: "https://swapi.co/api/vehicles/4/",
    vehicle_class: "wheeled"}
  },

  filteredVehicles() {
    return ["Sand Crawler", "Model: Digger Crawler", "Class: wheeled", "Passengers: 30", null, "2014-12-10T15:36:25.724000Z", false]
  }




}

export default AppData;