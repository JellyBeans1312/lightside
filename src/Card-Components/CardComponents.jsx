import React from 'react';
import Card from '../Card/Card.jsx'
import './CardComponents.scss'
import Home from '../Home/Home.jsx';

const CardComponents = ({group, addFavorite, page, crawl}) => {


  const cards = group.map(person => {
    return (<Card 
    props={person} 
    id={person[5]} 
    key={person[5]}
    addFavorite={addFavorite}
  />)})
      console.log('crawl', crawl)
      console.log(group.length)
      console.log(page)
  return (
    <section className='cardComponents'>
      {page === 'home' && <Home crawl={crawl}/>}
      {group.length !== 0 && cards}
      {group.length === 0 && page === 'favorites' 
        && <h2>You do not have any favorites</h2>
      }
    </section>
  )
}

export default CardComponents;