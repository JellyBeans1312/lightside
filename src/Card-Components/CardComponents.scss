
.cardComponents {
  height: 85%;
  display: flex;
  flex-wrap: wrap;
  h2 {
    color: white;
    margin: auto;
    margin-top: 50px;
  }
}