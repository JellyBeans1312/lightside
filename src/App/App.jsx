import React, { Component } from 'react';
import CardComponents from '../Card-Components/CardComponents';
import './App.scss';
import {Route} from "react-router-dom";
import Nav from '../Nav/Nav.jsx'
import Call from '../Fetch/FetchCall'


class App extends Component {
  constructor() {
    super()
    this.state = {
      film: {title:'', opening_crawl:'', release_date:''},
      people: [],
      vehicles: [],
      planets: [],
      allFavorites: [],
      isHidden: false
    }
  }
  
  componentDidMount = () => {
    if (window.location.href === 'http://localhost:3000/people') {
      this.showPeople()
    } else if (window.location.href === 'http://localhost:3000/planets') {
      this.getPlanets()
    } else if (window.location.href === 'http://localhost:3000/vehicles') {
      this.showVehicles()
    } else if (window.location.href === 'http://localhost:3000/') {
      this.showCrawl()
    }
  }

  showCrawl = () => {
    this.setState({film: {title:'', opening_crawl:'', release_date:''}})
      Call.fetchCrawl(this.getCrawl)
  }

  getCrawl = (crawl) => {
    this.setState({film: crawl})
  }

  showPeople = () => {
    if(this.state.people.length === 0) {
      Call.fetchPeople(this.getPeople)
    }
  }

  getPeople = (fetchData) => {
    const people = fetchData.map(person => {
      const info = [
        person.name, 
        `Birth Year: ${person.birth_year}`, 
        `Gender: ${person.gender}`, 
        `Height: ${person.height}`, 
        `Eye Color: ${person.eye_color}`, 
        person.created,
        false
      ]
      return info
    })
    this.setPeople(people)
    return people
  }

  setPeople = (people) => {
    this.setState({people: people})
  }

  showVehicles = () => {
    if(this.state.vehicles.length === 0) {
      Call.fetchVehicles(this.getVehicles)
    }
  }

    getVehicles = (fetchData) => {
      const vehicles = fetchData.map(vehicle => {
        const info = [
          vehicle.name,
          `Model: ${vehicle.model}`,
          `Class: ${vehicle.vehicle_class}`,
          `Passengers: ${vehicle.passengers}`,
          null,
          vehicle.created,
          false
        ]
        return info
      })
      this.setVehicles(vehicles)
      return vehicles
    }

    setVehicles = (vehicles) => {
      this.setState({vehicles: vehicles})
    }

    showPlanets = () => {
      if(this.state.planets.length === 0) {
        Call.fetchPlanets(this.getPlanets)
      }
    }
    
    getPlanets = (fetchData) => {
      const planets = fetchData.map(planet => {
        const info = [
          planet.name,
          `Terrain: ${planet.terrain}`,
          `Diameter: ${planet.diameter}`,
          `Population: ${planet.population}`,
          null,
          planet.created,
          false
        ]
        return info
      })
      this.setPlanets(planets)
      return planets
    }

    setPlanets = (planets) => {
      this.setState({planets: planets})
    }

  handleFavorite = (prop) => {
      const favorites = this.state.allFavorites;
      if(!favorites.includes(prop)) {
        prop[6] = !prop[6]
        this.setState({allFavorites: [...favorites ,prop]})
      } else {
        prop[6] = !prop[6]
        const unfavorite = favorites.filter(favorite => favorite !== prop)
        this.setState({allFavorites: unfavorite})
      }
  }

  render() {
    return (
      <main className="App">
        <Nav 
          getCrawl={this.showCrawl}
          getPeople={this.showPeople} 
          getPlanets={this.showPlanets}
          getVehicles={this.showVehicles}
          allFavorites={this.state.allFavorites}
        />
        <Route 
          exact path='/' 
          component={() => <CardComponents 
            group={this.state.people} 
            addFavorite={this.handleFavorite}
            crawl={this.state.film}
          />}
        />
        <Route 
          exact path='/people' 
          component={() => <CardComponents 
            group={this.state.people} 
            addFavorite={this.handleFavorite}
          />}
        />
        <Route 
          exact path='/planets' 
          component={() => <CardComponents 
            group={this.state.planets}
            addFavorite={this.handleFavorite}
          />}
        />
        <Route 
          exact path='/vehicles' 
          component={() => <CardComponents 
            group={this.state.vehicles}
            addFavorite={this.handleFavorite}
          />}
        />
        <Route 
          exact path='/favorites' 
          component={() => <CardComponents 
            group={this.state.allFavorites}
            addFavorite={this.handleFavorite}
          />}
        />
      </main>
    );
  }
}



export default App;
