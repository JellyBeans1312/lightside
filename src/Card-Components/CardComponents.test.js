import React from 'react';
import CardComponents from './CardComponents';
import {shallow} from 'enzyme';
import AppData from '../App/AppData';
import App from '../App/App.jsx'


describe('CardComponents', () => {
  let cardComponents, app ;
  beforeEach(() => {
    app = shallow(<App/>)
    cardComponents = shallow(<CardComponents
      group={[AppData.filteredPlanets()]}
      addFavorite={'Nathan'}
      crawl={{title:'Nathan', opening_crawl:'Nathan', release_date:'Nathan'}}
    />)
  })

  it('cardComponents to render in order', () => {
    expect(cardComponents).toMatchSnapshot()
  })


  // it('cards should return an array of arrays of cards', () => {
  //   expect(cardComponents).toEqual([])
  // })

})