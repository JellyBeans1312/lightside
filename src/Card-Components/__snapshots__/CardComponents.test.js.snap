// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponents cardComponents to render in order 1`] = `ShallowWrapper {}`;
